src/ripple/rpc/impl/ServerHandlerImp.cpp

```diff
@@ -182,20 +182,33 @@ ServerHandlerImp::onHandoff (Session& session,
     {
         if(is_ws)
         {
-            Handoff handoff;
-            auto const ws = session.websocketUpgrade();
+            std::shared_ptr<WSSession> ws;
```

Comment:
The `is_upgrade` block is the same in the two `onHandoff` functions. It would be a good cleanup if this were factored out into a helper function. No problem if you want to keep this PR just focused on the exceptions and save that cleanup for another PR.

Reply:
@seelabs Thanks, pushed cleanup.